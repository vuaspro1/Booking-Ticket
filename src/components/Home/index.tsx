/* eslint-disable @typescript-eslint/no-unused-vars */
import React, { FC } from 'react'
import { inject } from 'mobx-react'
import { observer } from 'mobx-react-lite'
import withLayout from '@src/lib/withLayout'
import { RootStoreHydration } from '@src/stores/RootStore'
import GlobalStore from '@src/stores/global.store'
import numeral from 'numeral'
import { Button, Carousel, Input, InputGroup, SelectPicker } from 'rsuite'
import RegisterBox from './RegisterBox'
import { Swiper, SwiperSlide } from 'swiper/react'
import { Autoplay } from 'swiper'

interface HomeBoxProps {
	globalStore?: GlobalStore
}

const HomeBox: FC<HomeBoxProps> = (props: HomeBoxProps) => {
	const { globalStore } = props

	return (
		<>
			<div className="c-bg-top">
				<div className="container">
					<Carousel autoplay className="custom-slider">
						<img src="https://metiz.vn/media/slide_home/gap-lai-chi-bau11.jpg" height="250" />
						<img src="https://metiz.vn/media/slide_home/Mai.jpg" height="250" />
						<img src="https://metiz.vn/media/slide_home/sieu_diep_vien.png" height="250" />
						<img src="https://metiz.vn/media/slide_home/dau_truong_muon_thu.png" height="250" />
						<img src="https://metiz.vn/media/slide_home/gia_dinh_diep_vien.jpg" height="250" />
					</Carousel>
					<div className="is-title">PHIM ĐANG CHIẾU</div>
					<Swiper
						slidesPerView={4}
						spaceBetween={30}
						loop={true}
						autoplay={{
							delay: 2000,
						}}
						modules={[Autoplay]}
						className="mySwiper"
					>
						<SwiperSlide>
							<img src="https://metiz.vn/media/slide_home/gap-lai-chi-bau11.jpg" height="250" />
						</SwiperSlide>
						<SwiperSlide>
							<img src="https://metiz.vn/media/slide_home/gap-lai-chi-bau11.jpg" height="250" />
						</SwiperSlide>
						<SwiperSlide>
							<img src="https://metiz.vn/media/slide_home/gap-lai-chi-bau11.jpg" height="250" />
						</SwiperSlide>
						<SwiperSlide>
							<img src="https://metiz.vn/media/slide_home/gap-lai-chi-bau11.jpg" height="250" />
						</SwiperSlide>
					</Swiper>
				</div>
			</div>
		</>
	)
}

export default withLayout(
	inject(({ store }: { store: RootStoreHydration }) => ({
		globalStore: store.globalStore,
		store,
	}))(observer(HomeBox))
)
