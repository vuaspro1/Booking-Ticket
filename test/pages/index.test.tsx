import { render /* , screen */ } from '@testing-library/react'
import '@testing-library/jest-dom'
import App from '../../src/pages/_app'
import Home from '../../src/pages/index'
/* import { createRouter } from 'next/router'

window.__NEXT_DATA__ = {
	props: {},
	page: '',
	query: {},
	buildId: '',
}

const router = createRouter('', { user: 'nikita' }, '', {
	subscription: jest.fn(),
	initialProps: {
		props: {},
	},
	pageLoader: jest.fn(),
	App: jest.fn(),
	Component: jest.fn(),
	wrapApp: jest.fn(),
	isFallback: false,
}) */
const useRouter = jest.spyOn(require('next/router'), 'useRouter')
const router: any = useRouter

describe('Home', () => {
	it('renders a heading', () => {
		const { container } = render(<App Component={Home} pageProps={{}} router={router} />)
		expect(container).toMatchSnapshot()

		/* const heading = screen.getByRole('heading', {
			name: /welcome to next\.js!/i,
		})

		expect(heading).toBeInTheDocument() */
	})
})
