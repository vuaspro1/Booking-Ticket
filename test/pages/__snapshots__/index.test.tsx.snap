// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders a heading 1`] = `
<div>
  <div
    class="c-page-loading"
  >
    <div
      class="c-loadingcs"
    >
      <div
        class="lds-spinner"
      >
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
      </div>
    </div>
  </div>
</div>
`;
